src/ucp/core/ucp_context.h

```diff
@@ -319,8 +319,7 @@ extern uct_memory_type_t ucm_to_uct_mem_type_map[];
 
 static UCS_F_ALWAYS_INLINE int ucp_memory_type_cache_is_empty(ucp_context_h context)
 {
-    return !(context->memtype_cache &&
-             context->memtype_cache->pgtable.num_regions);
+    return !context->memtype_cache->pgtable.num_regions;
```

Comment:
what is the reason to change this?

Reply:
moved if cache is "on" or "off" check to the caller in this PR so that it will disable ucp_tag_send_inline() if cache is "off".

Reply:
so this is adding few extra branches to fast path?

Reply:
I think number of checks are same as I removed  condition here and moved it up.  I don't see any perf difference for small messages in default case(mem type cache is ON).  On the other hand,  8 Byte HOST latency is improved from 1.4us to 1.05us because am_short in non fast path.